light/trie.go

```diff
@@ -80,14 +80,14 @@ func (db *odrDatabase) ContractCode(addrHash, codeHash common.Hash) ([]byte, err
 		return code, nil
 	}
 	id := *db.id
-	id.AccKey = addrHash[:]
+	id.AccKey = addr[:]
```

Comment:
Feels a bit strange to just change a byteslice representing a hash into representing an address. Just changing it doesn't mean it will work, does it?

Reply:
it won't work if the 'server' doesn't implement this change, but if both client and server do - and if I understand correctly - it will simply replace the hash of the data with the plain address in the request.

I checked that all the locations were updated accordingly, but given the errors in CI, it looks like I missed a point somewhere. I'll wait for Zsolt's input before I fix it, the best approach is still the entire removal of LES if it can be done.

Reply:
If we can fix the LES protocol, we can remove this account identifier(no matter it's address or hash) totally. Contract code is immutable and can be uniquely identified by hash.